### PR TITLE
Rule S1607: Tests should not be ignored

### LINKED ISSUE #662: Rule S1607: Tests should not be ignored
Implements [RSPEC-1607](https://jira.sonarsource.com/browse/RSPEC-1607)

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #662

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs [hunk 1]
```diff
@@ -82,39 +66,39 @@ public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<T> enumerable)
...
+        /// </summary>
+        /// <typeparam name="TFirst">The type of the elements of the first input sequence.</typeparam>
+        /// <typeparam name="TSecond">The type of the elements of the second input sequence.</typeparam>
+        /// <typeparam name="TResult">The type of the elements of the result sequence.</typeparam>
+        /// <param name="first">The first sequence to merge.</param>
+        /// <param name="second">The second sequence to merge.</param>
+        /// <param name="operation">A function that specifies how to merge the elements from the two sequences.</param>
+        /// <returns>An System.Collections.Generic.IEnumerable`1 that contains merged elements of
+        /// two input sequences.</returns>
+        public static IEnumerable<TResult> Merge<TFirst, TSecond, TResult>(this IEnumerable<TFirst> first,
```
Comment (SonarTech) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs]: MAJOR Reduce the number of generic parameters in the 'EnumerableExtensions.Merge' method to no more than the 2 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS2436)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs (hunk 1 above)
Comment (SonarTech) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs]: MAJOR Reduce the number of generic parameters in the 'EnumerableExtensions.Merge' method to no more than the 2 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS2436)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldNotBeIgnored.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class TestMethodShouldNotBeIgnored : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S1607";
+        private const string MessageFormat = "Either remove this 'Ignore' attribute or add an explanation about why " +
+            "it is skipped.";
```
Comment (valhristov) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldNotBeIgnored.cs]: perhaps should be `...an explanation about why this test is ignored.`

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs (hunk 1 above)
Comment (SonarTech) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs]: MAJOR Reduce the number of generic parameters in the 'EnumerableExtensions.Merge' method to no more than the 2 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS2436)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**